kernel: ignore z_fatal_halt() from code coverage

Intel-sp opened the pull request:
The reason we decide to ignore it in code coverage:
1.No test case can cover the function for code coverage.
2.Even if we added a test for testing, it would be marked as
  "never be called by other code" because the function cause
  CPU halted and it can't return.

Signed-off-by: Peng Su <peng.su@intel.com>

LeiW000 reviewed (approved):
It looks good to me. Please merge it. We needs it for code coverage testing.

Intel-sp committed:
kernel: ignore z_fatal_halt() from code coverage

The reason we decide to ignore it in code coverage:
1.No test case can cover the function for code coverage.
2.Even if we added a test for testing, it would be marked as
  "never be called by other code" because the function cause
  CPU halted and it can't return.

Signed-off-by: Peng Su <peng.su@intel.com>

LeiW000 commented:
@nashif , we want it merged before code freeze. Please help to take look at it. If there is no other comment, please merge it.
